include/RAJA/index/IndexSetBuilders.hpp

```diff
@@ -45,10 +56,165 @@ namespace RAJA
  *
  ******************************************************************************
  */
+template <typename WORKING_RES>
 void buildTypedIndexSetAligned(
     RAJA::TypedIndexSet<RAJA::RangeSegment, RAJA::ListSegment>& hiset,
     const Index_type* const indices_in,
-    Index_type length);
+    Index_type length)
```

Comment:
Should this take a resource as a template param or as an argument?
If it took an argument we could leave the implementations in the cpp.

Reply:
True. If you think that's better I can work up that change.

Reply:
I'm in favor of that. I think all of the planned public APIs take resources as arguments.

Reply:
OK. I will have those changes in and ready to review tomorrow morning.